analysis/R/decode_test.R

```diff
@@ -242,60 +249,106 @@ TestDecode <- function() {
   # match the ground truth. Must be close enough though.
   noise0 <- list(p = 0, q = 1, f = 0)  # no noise whatsoever
 
+  # Args are: message str, test function, # repetitions,
+  #           params, map, true pdf, # clients,
+  #           tolerances
   CheckDecodeAveAndStds("Testing Decode (1/5)", CheckDecodeHelper, 100,
-                        c(report4x2, noise0), map0, distribution0, 100,
+                        c(params_4x2, noise0), map0, distribution0, 100,
                         tolerance_l1 = 5,
                         tolerance_linf = 3)
 
   noise1 <- list(p = .4, q = .6, f = .5)  # substantial noise, very few reports
   CheckDecodeAveAndStds("Testing Decode (2/5)", CheckDecodeHelper, 100,
-                        c(report4x2, noise1), map0, distribution0, 100,
```

Comment:
For 2 reasons; to be consistent with Decode API and because in testing, GenerateCounts() simulates total counts using a map. I commented about this a few lines earlier but brought the comments down to this line.